README.md

```diff
@@ -374,6 +375,59 @@ render(
 );
 ```
 
+#### options.additionalParserRules - Parsing arbitrary patterns in markdown
+
+Pass the `options.additionalParserRules` property to the `compiler` or `<Markdown>` component to parse and replace arbitrary patterns in the markdown.
+
+You can use the enum `Priority` for order value.
+
+```jsx
+import Markdown from 'markdown-to-jsx';
+import React from 'react';
+
+// Matches a string that starts and end with `+`
+const customRegexMatcher =
+    /^([+])\1((?:\[.*?\][([].*?[)\]]|<.*?>(?:.*?<.*?>)?|`.*?`|~+.*?~+|.)*?)\1\1(?!\1)/;
+
+const md = `
+# Hello ++world++!
+`;
+
+render(
+    <Markdown
+        children={md}
+        options={{
+            additionalParserRules: {
+                example: {
+                    match: (source: string) => customRegexMatcher.exec(source),
+                    order: Priority.MED,
+                    parse: (capture, parse, state) => {
+                        return {
+                            content: parse(capture[2], state),
+                        }
+                    },
+                    react: (node, output, state) => {
+                        return <u key={state.key}>{output(node.content, state)}</u>
```

Comment:
One thing that occurred to me is markdown-to-jsx uses a custom JSX pragma to handle semantic element replacements and such. We would need to expose that `h(tag, props, children)` function to `react()` and the end result would look something more like this:

```js
react: (createElement, node, output, state) => {
  return createElement('u', {key: state.key}, output(node.content, state));
}
```

or perhaps

```js
react: (createElement, node, output, state) => {
  /* @jsx createElement */
  return <u key={state.key}>{output(node.content, state)}</u>
}
```

If jsx pragmas can be applied to particular block scopes.